packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_flutter_web.dart

```diff
@@ -305,11 +308,16 @@ class GoogleMapsPlugin extends GoogleMapsFlutterPlatform {
       polylines: polylines,
       circles: circles,
       mapOptions: mapOptions,
-    );
+    )..init(); // Initialize the controller
```

Comment:
This is the crux of the early initialization. Before the flow was:

buildView -> onPlatformViewCreated (plugin) -> init (plugin) -> init (controller) -> JS SDK

but now, it's:

buildView + init (controller) -> JS SDK -> onPlatformViewCreated (plugin) -> init (plugin, noop)